react/src/js/react/posts/PostView.jsx

```diff
@@ -1,25 +1,87 @@
 import React from "react";
 import Reflux from "reflux";
 
-import { Alert } from "react-bootstrap";
+import { Alert, Button, Form, FormGroup, FormControl, Thumbnail } from "react-bootstrap";
+import ReactMarkdown from "react-markdown";
 
 import { PostsStore, PostsActions } from "../discover/PostsStore";
 import LoadingComponent from "../misc/LoadingComponent";
+import { formatDate } from "../util/DateUtil";
 import Post from "./Post";
+import AuthStore from "../auth/AuthStore";
+import { Link } from "react-router-dom";
 
 /**
  * Renders details about one specific post
  */
 export default class PostView extends Reflux.Component {
     constructor(props) {
         super(props);
-        this.store = PostsStore;
+        this.stores = [PostsStore, AuthStore];
     }
 
     componentDidMount() {
         PostsActions.getPost(this.props.match.params.id);
     }
 
+    handleCommentSubmit = (e) => {
+        e.preventDefault();
+        if (e.currentTarget.elements.comment.value !== "") {
+            // eslint-disable-next-line no-undef
+            const formData = new FormData(),
+                form = e.currentTarget,
+                data = {
+                    comment: form.elements.comment.value,
+                    author: this.state.userId,
+                    contentType: form.elements.contentType.value,
+                    post: this.state.currentPost.id
+                };
+            formData.append("commentData", JSON.stringify(data));
+            formData.append("query", "addComment");
+            PostsActions.addComment(data);
+        }
+    }
+
+    renderCommentText(comment) {
+        if (comment.contentType === "text/markdown") {
+            return <ReactMarkdown source={comment.comment} />;
+        } else {
+            return <span>{comment.comment}</span>;
+        }
+    }
+
+    renderComments() {
+        const comments = this.state.currentPost.comments.map((comment) => (
+            <Thumbnail key={comment.id}>
+                <div className="comment">
+                    {this.renderCommentText(comment)} <br />
+                    Posted by <Link to={`/profile/${encodeURIComponent(comment.author.url)}/`}>{comment.author.displayName}</Link>  on {formatDate(comment.published)}
+                </div>
+            </Thumbnail>
+
+        ));
+        return comments;
+    }
+
+    renderMakeComments() {
+        return (
+            <Form onSubmit={this.handleCommentSubmit}>
+                <FormGroup controlId="comment">
+                    <FormControl bsSize="sm" name="comment" type="text" placeholder="Comment here..." />
```

Comment:
Shouldn't this be a textarea? Comments can be pretty large. With that in mind, you should also put a max length of something reasonable like 500 characters or something